### ISSUE TITLE
NodeJS `timers` module mock causing tests to timeout in JSDom

### ISSUE THREAD

**Issue by DiegoAndai** [sinonjs/fake-timers#470]
We have several tests timing out when updating `sinon` to `15.1.0`. After an initial investigation, the identified cause was `fake-timers` version `10.2.0`, specifically the [`timers` module mock change](https://github.com/sinonjs/fake-timers/pull/467), but don’t know why it caused the tests to fail.

You can check the PRs for `sinon` update here: [MUI PR](https://github.com/mui/material-ui/pull/37430), [MUI X PR](https://github.com/mui/mui-x/pull/9063). The failed tests are listed there.

- FakeTimers version: `10.2.0`
- Environment: JSDom
- Example URLs:
    - Material UI’s [Tooltip failing test](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Tooltip/Tooltip.test.js#L1036)
    - MUI X’s DataGridPro failing tests:
        - [Cell editing](https://github.com/mui/mui-x/blob/672a1c396a907f618ed8c1bc3772b7306c0141b3/packages/grid/x-data-grid-pro/src/tests/cellEditing.DataGridPro.test.tsx#L55)
        - [Print Export](https://github.com/mui/mui-x/blob/672a1c396a907f618ed8c1bc3772b7306c0141b3/packages/grid/x-data-grid-pro/src/tests/printExport.DataGrid.test.tsx#L79)
        - [Row Editing](https://github.com/mui/mui-x/blob/672a1c396a907f618ed8c1bc3772b7306c0141b3/packages/grid/x-data-grid-pro/src/tests/rowEditing.DataGridPro.test.tsx#L88)
- Other libraries you are using: Don’t know which would be relevant, let me know.

**What did you expect to happen?**

Tests were running ok before `fake-timers` version `10.2.0`, version `10.1.0` works fine.

**What actually happens**

Several tests listed are timing out, for example, the ones listed above.

**How to reproduce**

A minimal test for our codebase that passes before the upgrade and fails after:

```jsx
it('test', async () => {
  await act(() => {
    return new Promise((resolve) => {
      setTimeout(resolve, 100);
      clock.runAll();
    });
  });
  expect(true).to.equal(true);
});
```

Fails with the following error:

```bash
Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
  at listOnTimeout (internal/timers.js:557:17)
  at processTimers (internal/timers.js:500:7)
```

The other examples listed above are failing with the same error.

Thanks in advance for checking this out! 😊

**Comment by fatso83** [comment]
Seems like the new native Node timers feature broke some installs.

cc @swenzel-arc and @benjamingr 

I think we might want to deprecate this version and re-release as a new major to avoid this while we figure out the details of what is happening.

@DiegoAndai I wrote you directly in one of the PRs you referenced: https://github.com/mui/material-ui/pull/37430 I analyzed the situation a bit to figure out the likely suspects. I do not have a fix without looking more into it, but suggested something to look at.

**Comment by fatso83** [comment]
Just a rough summary to what has been done (see  https://github.com/mui/material-ui/pull/37430 for details):
- verified that the MUI issue was reproducible (does not happen in 10.1, but does happen in 10.2)
- did not see _what_ causes the issue specifically (requires work to be put in)
- deprecated NPM package version 10.2
- released a new major version 11 - no functional changes, but to signal a breaking change
- released 10.3 as a re-tagged 10.1 to fix Sinon installs that still chose to install 10.2
- released new version of Sinon 15.2 that explicitly relies on 10.3 (to avoid the 10.2 issue)

If we depend on version 11 in Sinon, that will be a breaking change.

The reason the linked PR still fails is due to an unrelated change that ironically was to improve webpack compatibility: https://github.com/sinonjs/sinon/pull/2519

I don't think this change validates a new major version, as we usually care about API changes, not bundling compatibility, with regards to major releases.

**Comment by fatso83** [comment]
Closing this as I have no idea what is wrong and if something is wrong, and there exists a workaround.
